Update the intro and appendix (#62)
riscv-non-isa/riscv-external-debug-security: intro.adoc

This specification defines non-ISA extension <<dmsext, Debug Module Security Extension (non-ISA extension)>> and ISA extension <<Sdsec, Sdsec (ISA extension)>> to address the above security issues in the current _The RISC-V Debug Specification_ cite:[dbgspec] and trace specifications cite:[etrace] cite:[ntrace]. 

A summary of the changes introduced by _The RISC-V External Debug Security Specification_ follows.:

    - *Per-Hart Debug Control:* Introduce per-hart states to control whether external debug is allowed in M-mode and/or supervisor domains cite:[smmtt].
    - *Per-Hart Trace Control:* Introduce per-hart states to control whether tracing is allowed in M-mode and/or supervisor domains.
    - *Non-secure debug:* Add a non-secure debug state to relax security constraints.
    - *Debug Mode entry:* External debugger can only halt the hart and enter debug mode when debug is allowed in current privilege mode; all operations are executed with <<dbgaccpriv, debug access privilege>> instead of M-mode privilege.

>     - *Memory Access:* Memory access from a hart’s point of view using a Program Buffer or the Abstract Command must be checked by the hart's memory protection mechanisms as if the hart is running at <<dbgaccpriv, debug access privilege>>; memory access from Debug Module using System Bus Access block without involving a hart must be checked by system memory protection mechanism, such as IOPMP or WorldGuard.

Reviewer: > a Program Buffer or the Abstract Command

I would say "the Program Buffer or an Abstract Command"

Reviewer: > from Debug Module

I would say "from the Debug Module"

Reviewer 2: It took me a few minutes to understand the part after the semicolon.  I think that it should be:
```suggestion
    - *Memory Access:* Memory access from a hart’s point of view using the Program Buffer or an Abstract Command must be checked by the hart's memory protection mechanisms as if the hart is running at <<dbgaccpriv, debug access privilege>>; memory access from the Debug Module using System Bus Access must be checked by a system memory protection mechanism, such as IOPMP or WorldGuard.
```
SBA is defined to not use a hart.  Either the above or "memory access from the Debug Module using System Bus Access without involving a hart..."  My main confusion was that "block" can be either a verb or a noun and SBA is architecturally more of a feature than a block so my brain tried to parse it as a verb.

Author: Sorry for the confusion 😂. Will take your suggestion

>     - *Register Access:* Register access using Program Buffer or the Abstract Command works as if the hart is running in <<dbgaccpriv, debug access privilege>> instead of M-mode privilege. The debug CSRs (`dcsr` and `dpc` ) are shadowed in supervisor domains while Smtdeleg/Sstcfg extensions expose the trigger CSRs to supervisor domains through indirect CSR access. 

Reviewer 2: ```suggestion
    - *Register Access:* Register access using the Program Buffer or an Abstract Command works as if the hart is running in <<dbgaccpriv, debug access privilege>> instead of M-mode privilege. The debug CSRs (`dcsr` and `dpc` ) are shadowed in supervisor domains while Smtdeleg/Sstcfg extensions expose the trigger CSRs to supervisor domains through indirect CSR access. 
```
